data_structures/rb_tree2.cpp

```diff
@@ -0,0 +1,1628 @@
+            // sibling is a right child of its parent
+
+            // right right: with right or 2 red kids
+            if (s_right != nullptr && s_right->color == COLOR_RED) {
+                // rotate left once and recolor
+                RotateLeft(sibling);
+                ShiftColorsRight(sibling);
+            } else {
+                // right left: with 1 red kid on left
+
+                // rotate right on child, making a right right rotate case
+                RotateRight(s_left);
+                sibling = s_left;
+                ColorSwap(sibling, sibling->right);
+
+                // rotate left and recolor to fix new right right case
+                RotateLeft(sibling);
+                ShiftColorsRight(sibling);
+            }
+        }
+        // color level returns to black
+        replacement->color--;
+    }
+
+    /**
+     * @brief Shifts colors of a node and its children to the right
+     * @param sibling the node where color shift occurs
+     */
+    void ShiftColorsRight(RBTNode* sibling) {
+        sibling->right->color = sibling->color;
+        sibling->color = sibling->left->color;
+        sibling->left->color = COLOR_BLACK;
+    }
+
+    /**
+     * @brief Shifts colors of a node and its children to the left
+     * @note
+     * A left color shift is necessary after a right rotation occurs
+     * in the red black tree.
+     * @param sibling the node where color shift occurs
+     */
+    void ShiftColorsLeft(RBTNode* sibling) {
+        sibling->left->color = sibling->color;
+        sibling->color = sibling->right->color;
+        sibling->right->color = COLOR_BLACK;
+    }
+
+    /**
+     * @brief Gives the pointer from a node to its only child
+     * @details
+     * Assuming node 'n' has only one valid child,
+     * this function checks if it's a left or right child and
+     * returns the pointer to it from parent node 'n'
+     * (ex. returns 'n->left' or 'n->right')
+     * @param n the node to get the child pointer for
+     * @return RBTNode*& the node's pointer to its only child
+     */
+    RBTNode*& MeToOnlyChildPtr(RBTNode* n) {
+        // check node has exactly one non-null child
+        if ((n->left == nullptr || n->right == nullptr) &&
+            !(n->left == nullptr && n->right == nullptr)) {
+            // is left child
+            if (n->left != nullptr) {
+                return n->left;
+            } else if (n->right != nullptr) {
+                // is right child
+
+                return n->right;
+            }
+        }
+        // reaches here if passed node is invalid for function
+        throw invalid_argument(
+            "MeToOnlyChildPtr(): Node passed "
+            "doesn't have exactly one child.");
+    }
+
+    /**
+     * @brief Gives the pointer from a node's parent to itself
+     * @param n the node pointed to by its parent
+     * @return RBTNode*& the parent's pointer to the node
+     */
+    RBTNode*& ParentToMePtr(RBTNode* n) {
```

Comment:
could you  please follow snake case?
ie
this_is_a_name_in_snake_case - snake case
YouAreUsingCamelCase - camel case

the repo uses camel case

Reply:
Yeah I'll update those, I wasn't sure if I should since the contributing guidelines only mentioned having the file name in snake case but I get it for consistency